### PR TITLE
optionally pass plugin conf path to persistence plugins, #19822

### PR DESCRIPTION
**Pull request by patriknw**
To unlock https://github.com/akka/akka-persistence-cassandra/issues/81

Refs #19822

### CODE REVIEW

File: akka-docs/rst/scala/persistence.rst [hunk 1]
```diff
@@ -937,15 +937,21 @@ A journal plugin can be activated with the following minimal configuration:
...
 are executed sequentially. It may delegate to asynchronous libraries, spawn futures, or delegate to other
 actors to achive parallelism.
 
-The journal plugin class must have a constructor without parameters or a constructor with one ``com.typesafe.config.Config``
-parameter. The plugin section of the actor system's config will be passed in the config constructor parameter.
+The journal plugin class must have a constructor with one of these signatures:
+
+* constructor with one ``com.typesafe.config.Config`` parameter and a ``String`` parameter for the config path
+* constructor with one ``com.typesafe.config.Config`` parameter
+* constructor without parameters
```
Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: Isn't the ExtendedActorSYstem always expected to be a param?

`DummyReadJournalProvider4(sys: ExtendedActorSystem, conf: Config, confPath: String) extends `

?

Comment (patriknw) [inline akka-docs/rst/scala/persistence.rst]: it also supports no params as can be seen here https://github.com/akka/akka/pull/22000/files#diff-c44f1300bede3f99280e3aa647346d3aR92

Comment (ktoso) [inline akka-docs/rst/scala/persistence.rst]: Yeah, though I think the points should mention it should accept `actorSystem, config, string` not just the `config, string` part?

Comment (patriknw) [inline akka-docs/rst/scala/persistence.rst]: ah, good point copy pasta (journal plugin doesn't take system)

**Pull request merged**